core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala

```diff
@@ -0,0 +1,268 @@
+                          creationId: CreationId,
+                          isBlackboxInvocation: Boolean) = {
+    creationJobPool getOrElseUpdate (creationId, {
+      val key = inProgressContainer(invocationNamespace, action, revision, schedulerInstanceId, creationId)
+      dataManagementService ! RegisterData(key, "", failoverEnabled = false)
+      JobEntry(action, createTimer(invocationNamespace, action, revision, creationId, isBlackboxInvocation))
+    })
+  }
+
+  private def deleteJob(invocationNamespace: String,
+                        action: FullyQualifiedEntityName,
+                        revision: DocRevision,
+                        creationId: CreationId,
+                        state: CreationJobState) = {
+    val key = inProgressContainer(invocationNamespace, action, revision, schedulerInstanceId, creationId)
+
+    // If there is a JobEntry, delete it.
+    creationJobPool
+      .remove(creationId)
+      .foreach(entry => {
+        sendState(state)
+        entry.timer.cancel()
+      })
+
+    dataManagementService ! UnregisterData(key)
+    Future.successful({})
+  }
+
+  private def sendState(state: CreationJobState): Unit = {
+    context.parent ! state // send state to ContainerManager
+    QueuePool.get(MemoryQueueKey(state.invocationNamespace, state.action.toDocId.asDocInfo(state.revision))) match {
+      case Some(memoryQueueValue) if memoryQueueValue.isLeader =>
+        memoryQueueValue.queue ! state
+      case _ =>
+        logging.error(this, s"get a $state for a nonexistent memory queue or a follower")
+    }
+  }
+
+  protected def createTimer(invocationNamespace: String,
+                            action: FullyQualifiedEntityName,
+                            revision: DocRevision,
+                            creationId: CreationId,
+                            isBlackbox: Boolean): Cancellable = {
+    val timeout = if (isBlackbox) FiniteDuration(baseTimeout.toSeconds * 3, TimeUnit.SECONDS) else baseTimeout
+    actorSystem.scheduler.scheduleOnce(timeout) {
+      logging.warn(
+        this,
+        s"Failed to create a container for $action(blackbox: $isBlackbox), error: $creationId timed out after $timeout")
+      creationJobPool
+        .remove(creationId)
+        .foreach(
+          _ =>
+            sendState(
+              FailedCreationJob(
+                creationId,
+                invocationNamespace,
+                action,
+                revision,
+                ContainerCreationError.TimeoutError,
+                s"timeout waiting for the ack of $creationId after $timeout")))
+      dataManagementService ! UnregisterData(
+        inProgressContainer(invocationNamespace, action, revision, schedulerInstanceId, creationId))
+    }
+  }
+
+  private val topicPrefix = loadConfigOrThrow[String](ConfigKeys.kafkaTopicsPrefix)
+  private val topic = s"${topicPrefix}creationAck${schedulerInstanceId.asString}"
+  private val maxActiveAcksPerPoll = 128
```

Comment:
In Scheduler.scala, this value is configurable.
Can we make it configurable as well?

Reply:
it seems other places are also hardcoded